### PR TITLE
updates to ArgParser - added smart_argument_value(...) function

### LINKED ISSUE #449: Add Argparser type casting
The argparser is located here : https://github.com/ObEngine/ObEngine/blob/v0.5/src/Core/Utils/ArgParser.cpp#L7

The improvements would be the following one :
- Detect numbers (int and float) and cast them to appropriate vili type
- Detect `true` and `false` and cast them to booleans
- Detect flags without values (`--enable-this --enable-that`) and cast them to `true` (boolean)
- Quoting a value will force it to be a string but non-quoted values are also fine (except of course if you want a string equal to "true" for example)

### PR DESCRIPTION
**Pull request by DarthData410**
Based on requirements in issue #449 this PR contains changes to address the listed enhancement request.

### CODE REVIEW

File: src/Core/Utils/ArgParser.cpp [hunk 1]
```diff
@@ -4,6 +4,72 @@
 
 namespace obe::utils::argparser
 {
+    /// @brief Test string to see if it is a bool value
```
Comment (Sygmei) [inline src/Core/Utils/ArgParser.cpp]: Please use this style for Doxygen comments :) https://github.com/ObEngine/ObEngine/blob/c3095b1c2cb3fdcc2a2df04aae3052fb3cc1a5dd/include/Core/Animation/Animation.hpp#L121

Comment (Sygmei) [reply src/Core/Utils/ArgParser.cpp]: (Also, while it's nice that you took the time to do it, for next time, no need to document private/internal functions, especially for functions with a name as clear as those)

File: src/Core/Utils/ArgParser.cpp [hunk 2]
```diff
@@ -4,6 +4,72 @@
...
+    }
+
+    /// @brief Converts string value representing argv[idx] into expected C-type, before converting to vili type
+    /// @param argvalue string value representing arguvment value to-be converted
+    /// @return vili::node representing converted value from string to C-type/vili type
+    vili::node smart_argument_value(std::string argvalue)
+    {
+        vili::node ret = argvalue;
+
+        if(std::isalpha(argvalue[0])) {
```
Comment (Sygmei) [inline src/Core/Utils/ArgParser.cpp]: Please follow [`.clang-format`](https://github.com/ObEngine/ObEngine/blob/v0.5/.clang-format) codestyle :)
Don't hesitate if you need some help to set it up !

**Comment by Sygmei** [comment]
Apart from the codestyle issue, this is great.
With this PR and your previous one, using args in ÖbEngine will be much easier 👍

**Comment by DarthData410** [comment]
Alright @Sygmei I will get the desired changes in place, update ad recommit. Glad to be contributing, and I look forward to doing more.

File: src/Core/Utils/ArgParser.cpp (hunk 1 above)
Comment (DarthData410) [reply src/Core/Utils/ArgParser.cpp]: Just removing comments from internal functions for ArgParser.

File: src/Core/Utils/ArgParser.cpp (hunk 2 above)
Comment (DarthData410) [reply src/Core/Utils/ArgParser.cpp]: @Sygmei I believe all the code formatting requested changes are in place. Also removed the comments for these internal functions. Let me know if there is anything else for this enhancement / PR.

**Comment by Sygmei** [comment]
Awesome ! Thanks for the contribution :)
Everything looks good, since there is a merge conflict I'll take a look at it in a few hours when I'm home !

**Comment by Sygmei** [comment]
Merged ! Thanks again for your contribution !

Little tip for next contribution : Create a branch on your fork with a nice name for the feature you work on

eg: `git checkout -b feat/argparser_typecasting`

**Pull request merged**